File: freeipa/src/main/java/com/sequenceiq/freeipa/entity/json/JsonToString.java
package com.sequenceiq.freeipa.entity.json;

import javax.persistence.AttributeConverter;

public class JsonToString implements AttributeConverter<Json, String> {
    @Override
    public String convertToDatabaseColumn(Json attribute) {
        if (attribute != null) {
            return attribute.getValue();
        }
        return null;
    }

    @Override
    public Json convertToEntityAttribute(String dbData) {
        return new Json(dbData);
    }
}

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/json/JsonToString.kt
package com.sequenceiq.cloudbreak.domain.json

import javax.persistence.AttributeConverter

class JsonToString : AttributeConverter<Json, String> {
    override fun convertToDatabaseColumn(attribute: Json?): String? {
        if (attribute != null) {
            return attribute.value
        }
        return null
    }

    override fun convertToEntityAttribute(dbData: String): Json {
        return Json(dbData)
    }
}
